### PR TITLE
Build project on MVVM Architecture

### PR DESCRIPTION
**Pull request by mihaidornea**

### CODE REVIEW

File: app/src/main/java/com/example/wd_mvvm/ui/news/NewsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        newsViewModel?.init()
+    }
+
+    private fun getNews() {
+        newsViewModel!!.getNews()?.observe(this, Observer<NewsApiResponse>{
+            displayNews(it.articles)
+        })
+    }
+
+    private fun loadUIElements() {
```
Comment (andiwolfpack) [inline app/src/main/java/com/example/wd_mvvm/ui/news/NewsFragment.kt]: You can make a BindingAdapter for this and set the adapter in the xml as viewModel.adapter. Then you can handle all the adapter updates in the viewmodel

File: app/src/main/java/com/example/wd_mvvm/ui/news/NewsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        super.onStart()
+        getNews()
+    }
+
+    private fun getViewModel() {
+        newsViewModel = ViewModelProviders.of(this).get(NewsViewModel::class.java)
+        newsViewModel?.init()
+    }
+
+    private fun getNews() {
```
Comment (andiwolfpack) [inline app/src/main/java/com/example/wd_mvvm/ui/news/NewsFragment.kt]: If you move the adapter in the viewmodel you don't need this

**Review by andiwolfpack — CHANGES_REQUESTED** [review]

**Review by andiwolfpack — APPROVED** [review]

File: app/src/main/java/com/example/wd_mvvm/models/NewsApiResponse.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.example.wd_mvvm.models
+
+import com.example.wd_mvvm.models.article.Article
+
+data class NewsApiResponse (
+    var status: String,
+    var totalResults: Int,
```
Comment (mgabbi) [inline app/src/main/java/com/example/wd_mvvm/models/NewsApiResponse.kt:10]: Take care to specify the serialized names in the future using @SerializedName for every field

File: app/src/main/java/com/example/wd_mvvm/repository/Repository.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import org.koin.core.inject
+import retrofit2.Call
+import retrofit2.Callback
+
+class Repository : KoinComponent {
+
+    private val remoteDataSource by inject<RemoteDataSourceImpl>()
+
+    fun getNews(sources: String, callback: Callback<NewsApiResponse>) {
+        return remoteDataSource.getNews(sources, callback)
```
Comment (mgabbi) [inline app/src/main/java/com/example/wd_mvvm/repository/Repository.kt]: You don't need a return here

**Review by mgabbi — CHANGES_REQUESTED** [review]

**Pull request merged**